gpustack/detectors/detector_factory.py

```diff
@@ -0,0 +1,75 @@
+import logging
+from typing import Dict
+from gpustack.detectors.base import (
+    GPUDetector,
+    GPUDevicesInfo,
+)
+from gpustack.schemas.workers import SystemInfo, VendorEnum
+from gpustack.detectors.fastfetch.fastfetch import Fastfetch
+from gpustack.detectors.npu_smi.npu_smi import NPUSMI
+
+
+logger = logging.getLogger(__name__)
+
+
+class DetectorFactory:
+    def __init__(
+        self,
+        gpu_detectors: Dict[str, GPUDetector] = None,
```

Comment:
How do we use this param? I don't find reference

Reply:
This is for user-defined worker resources